### PR TITLE
Use PVs for mons and update deployment example

### PR DESCRIPTION
**Pull request by jarrpa**
This PR supersedes #23 

Signed-off-by: Jose A. Rivera <jarrpa@redhat.com>

### CODE REVIEW

File: deploy/deploy-with-olm.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ metadata:
   namespace: openshift-marketplace
 spec:
   sourceType: grpc
-  image: quay.io/ocs-dev/ocs-registry:latest
+  image: quay.io/kshlm/ocs-registry:mon-test
```
Comment (obnoxxx) [inline deploy/deploy-with-olm.yaml]: Why is this? This shouldn't be pushed to the official repo...

File: deploy/olm-catalog/ocs-operator/0.0.1/ocs-operator.v0.0.1.clusterserviceversion.yaml [hunk 2]
```diff
@@ -205,7 +205,7 @@ spec:
                       fieldPath: metadata.name
                 - name: OPERATOR_NAME
                   value: ocs-operator
-                image: quay.io/ocs-dev/ocs-operator:latest
+                image: quay.io/kshlm/ocs-operator:mon-test
```
Comment (obnoxxx) [inline deploy/olm-catalog/ocs-operator/0.0.1/ocs-operator.v0.0.1.clusterserviceversion.yaml]: Why is this? This shouldn't be pushed to the official repo...

File: pkg/controller/storagecluster/reconcile.go [hunk 3]
```diff
@@ -101,6 +102,17 @@ func newCephCluster(sc *ocsv1alpha1.StorageCluster) *rookCephv1.CephCluster {
 			Mon: rookCephv1.MonSpec{
 				Count:                3,
 				AllowMultiplePerNode: false,
+				VolumeClaimTemplate: &corev1.PersistentVolumeClaim{
+					Spec: corev1.PersistentVolumeClaimSpec{
+						// TODO: Don't hardcode StorageClassName
+						StorageClassName: strptr("gp2"),
```
Comment (obnoxxx) [inline pkg/controller/storagecluster/reconcile.go]: Where does that StorageClassName come from? Where *should* it come from?

File: pkg/controller/storagecluster/reconcile.go [hunk 4]
```diff
@@ -101,6 +102,17 @@ func newCephCluster(sc *ocsv1alpha1.StorageCluster) *rookCephv1.CephCluster {
...
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceStorage: resource.MustParse("10Gi"),
```
Comment (obnoxxx) [inline pkg/controller/storagecluster/reconcile.go]: Is that a standard size for the mon's backing store volume?

File: deploy/crds/ocs_v1alpha1_storagecluster_cr.yaml [hunk 5]
```diff
@@ -5,3 +5,17 @@ metadata:
...
 spec:
   manageNodes: false
+  storageDeviceSets:
+  - name: example-deviceset
+    count: 3
+    resources: {}
+    placement: {}
+    dataPVCTemplate:
+      spec:
+        storageClassName: gp2
```
Comment (obnoxxx) [inline deploy/crds/ocs_v1alpha1_storagecluster_cr.yaml]: OK. I see... this is where the `gp2` name has its origin. Somewhat weird to have it defined in an example (is it?), and referenced hard in the code..

**Comment by kshlm** [comment]
I never meant for #23 to be merged in the way it was structured. And as this PR is mainly a rebase, with the commit messages updated, I don't want this to be merged either.

I've opened #30 as the successor to #23. It contains the changes to the deploy manifest and CSV, which were found in #23. The change to get mons on pvs will come in another PR on top of #30.

I'd like this to be closed and #30 be reviewed instead.

**Pull request closed without merging**